pkg/operator/staticpod/controller/installer/installer_controller.go

```diff
@@ -1091,6 +1086,17 @@ func (c InstallerController) Sync(ctx context.Context, syncCtx factory.SyncConte
 	return err
 }
 
+func deepCopyNodeStatusWithoutOldFailedState(ns *operatorv1.NodeStatus) *operatorv1.NodeStatus {
+	if ns.TargetRevision == 0 || ns.TargetRevision != ns.LastFailedRevision {
+		return &operatorv1.NodeStatus{
```

Comment:
are we going to overwrite the failed state in this case?

Reply:
yes, that the "we don't track history of fails" part. I call this func when we set a new set of `lastFailed*` fields.